main/ejml-dsparse/src/org/ejml/sparse/csc/misc/ImplCommonOpsWithSemiRing_DSCC.java

```diff
@@ -146,9 +158,10 @@ public static void elementMult( DMatrixSparseCSC A, DMatrixSparseCSC B, DMatrixS
 
             // compute the maximum number of elements that there can be in this row
             int maxInRow = Math.min(idxA1 - idxA0, idxB1 - idxB0);
+            int expectedResultSize = C.nz_length + maxInRow;
 
             // make sure there are enough non-zero elements in C
-            if (C.nz_length + maxInRow > C.nz_values.length)
+            if (expectedResultSize > C.nz_values.length && expectedResultSize < maxMaskEntries)
```

Comment:
I've not gone through the logic here carefully, but this jumps out as a potential issue. Are there unit tests which would trigger the logic on the right and have you checked edge conditions?

Reply:
I have the dedicated test `TestImplCommonOpsWithSemiRing_DSCC#useMaskForResultExpansionInMult` for this condition.
The idea is, the mask provides a __pessimistic__ estimation of its set entries. Thereby the number of set entries corresponds to the largest possible result size. The estimation will never be smaller than the actual number of set entries. 
So with this check we avoid growing larger than necessary.

Explaining the check, I also see this condition is hard to read. I will simplify the check :+1:

Reply:
thanks! easier to understand now.

Reply:
If this growth logic occurs in several locations we should probably create a static function that does it and create a new PR for it.

Reply:
Yes I think this could be beneficial when we add more sparse operations with mask support